docs/en/Running-Alluxio-On-Kubernetes.md

```diff
@@ -0,0 +1,124 @@
+---
+layout: global
+title: Running Alluxio on Kubernetes
+nickname: Alluxio on Kubernetes
+group: Deploying Alluxio
+priority: 3
+---
+
+Alluxio can be run on Kubernetes. This guide demonstrates how to run Alluxio
+on Kubernetes using the specification that comes in the Alluxio Github repository.
+
+# Basic Tutorial
+
+This tutorial walks through a basic Alluxio setup on Kubernetes.
+
+## Prerequisites
+
+- A Kubernetes cluster (version >= 1.8). Alluxio workers will use `emptyDir` volumes with a 
+restricted size using the `sizeLimit` parameter. This is an alpha feature in Kubernetes 1.8.
+Please ensure the feature is enabled.
+- An Alluxio Docker image. Refer to [this page](Running-Alluxio-On-Docker.html) for instructions
+to build an image. The image must be available for a pull from all Kubernetes hosts running
+Alluxio processes. This can be achieved by pushing the image to an accessible docker registry,
+or pushing the image individually to all hosts. If using a private docker registry, refer to the
+Kubernetes [documentation](https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/).
+
+## Clone the Alluxio repo
+
+```bash
+$ git clone https://github.com/Alluxio/alluxio.git
+$ cd integration/kubernetes
+```
+
+The kubernetes specifications required to deploy Alluxio can be found under `integration/kubernetes`.
+
+## Enable short-circuit operations
+
+Short-circuit access enables clients to perform read and write operations directly against the 
+worker memory instead of having to go through the worker process. Set up a domain socket on all hosts
+eligible to run the Alluxio worker process to enable this mode of operation.
+
+From the host machine, create a directory for the shared domain socket.
+```bash
+$ mkdir /tmp/domain
+$ chmod a+w /tmp/domain
+$ touch /tmp/domain/d
+$ chmod a+w /tmp/domain/d
+```
+
+This step can be skipped in case short-circuit accesss is not desired or cannot be set up. To disable
+this feature, set the property `alluxio.user.short.circuit.enabled=false` according to the instructions
+in the configuration section below.
+
+## Provision a Persistent Volume
+
+Alluxio master can be configured to use a persistent volume for storing the journal. The volume,
+once claimed, is persisted across restarts of the master process. Prepare a persistent volume on 
+hosts eligible to run the Alluxio master process.
```

Comment:
if each host has its own persistent volume, does that mean that if kubernetes restarts the master on a different host, it will start from a new journal?

Reply:
a persistent volume can be host-based or not. nfs volumes or ebs volumes for example are not host based volumes. the example I provide is (hostPath), which means a pod can only be rescheduled on that one host (it is bound). 

but you bring up a good point, i'll add more docs for setting up the persistent volume. thanks.

Reply:
thanks for the explanation, glad to know k8s is smart about it